src/Storages/PostgreSQL/PostgreSQLReplicationHandler.cpp

```diff
@@ -186,6 +186,8 @@ void PostgreSQLReplicationHandler::startSynchronization(bool throw_on_error)
             auto * materialized_storage = storage->as <StorageMaterializedPostgreSQL>();
             try
             {
+                /// FIXME: Looks like it is possible we might get here if there is no nested storage or at least nested storage id field might be empty.
```

Comment:
Very scary.

Reply:
Not very scary because "nested storage id might be empty" means not that there is no StorageID in some storage, but that a filed `nested_table_id` in MaterializedPostgreSQL storage is std::nullopt.
Is is set in 2 cases:
1. At constructor -- if MaterializedPostgreSQL storage is created with a ready fully made nested storage passed into its constrcutor.
2. in storage->prepare() call -- right after materialized storage has created and saved nested storage inside itself.

I'll add this to a comment.